libs/components/popovers/testing/src/dropdown/harness/dropdown-harness.spec.ts

```diff
@@ -0,0 +1,283 @@
+
+    await expectAsync(dropdownHarness.isOpen()).toBeResolvedTo(false);
+
+    await dropdownHarness.clickDropdownButton();
+    fixture.detectChanges();
+
+    await expectAsync(dropdownHarness.isOpen()).toBeResolvedTo(true);
+
+    await dropdownHarness.clickDropdownButton();
+    fixture.detectChanges();
+
+    await expectAsync(dropdownHarness.isOpen()).toBeResolvedTo(false);
+  });
+
+  it('should get the dropdown menu harness', async () => {
+    const { dropdownHarness, fixture } = await setupTest();
+
+    fixture.componentInstance.menuRole = 'dropdown-menu';
+    fixture.detectChanges();
+    await dropdownHarness.clickDropdownButton();
+    fixture.detectChanges();
+
+    const dropdownMenuHarness = await dropdownHarness.getDropdownMenu();
+
+    await expectAsync(dropdownMenuHarness?.getRole()).toBeResolvedTo(
+      'dropdown-menu'
+    );
+  });
+
+  describe('Dropdown menu test harness', () => {
+    it('should get throw an error if dropdown menu is not open', async () => {
+      const { dropdownHarness, fixture } = await setupTest();
+
+      fixture.detectChanges();
+
+      await expectAsync(
+        dropdownHarness.getDropdownMenu()
+      ).toBeRejectedWithError(
+        'Unable to retrieve dropdown menu harness because dropdown is closed'
+      );
+    });
+
+    it('should throw an error when trying to get the role of an item in an empty menu', async () => {
+      const { dropdownHarness, fixture } = await setupTest();
+
+      fixture.detectChanges();
+      await dropdownHarness.clickDropdownButton();
+      fixture.detectChanges();
+
+      const dropdownMenuHarness = await dropdownHarness.getDropdownMenu();
+
+      await expectAsync(
+        dropdownMenuHarness?.getItemRoleAtIndex(0)
+      ).toBeRejectedWithError(
+        'Unable to retrieve item because dropdown menu is empty'
+      );
+    });
+
+    it('should throw an error when trying to click an item at index in an empty menu', async () => {
+      const { dropdownHarness, fixture } = await setupTest();
+
+      fixture.detectChanges();
+      await dropdownHarness.clickDropdownButton();
+      fixture.detectChanges();
+
+      const dropdownMenuHarness = await dropdownHarness.getDropdownMenu();
+
+      await expectAsync(
+        dropdownMenuHarness?.clickItemAtIndex(0)
+      ).toBeRejectedWithError(
+        'Unable to retrieve item because dropdown menu is empty'
+      );
+    });
+
+    it('should throw an error when trying to click an item with role in an empty menu', async () => {
+      const { dropdownHarness, fixture } = await setupTest();
+
+      fixture.detectChanges();
+      await dropdownHarness.clickDropdownButton();
+      fixture.detectChanges();
+
+      const dropdownMenuHarness = await dropdownHarness.getDropdownMenu();
+
+      await expectAsync(
+        dropdownMenuHarness?.clickItemWithRole('item-role')
+      ).toBeRejectedWithError(
+        'Unable to retrieve item because dropdown menu is empty'
+      );
+    });
```

Comment:
moved `DropdownMenuHarness` testing on empty menu in dropdown unit test instead of dropdown-menu unit tests because different menus can only be accessed from the `DropdownHarness` and not by `dataSkyId`